pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -589,20 +605,59 @@ def deserialize_abstract_layout(obj_str: str) -> RegisterLayout:
     return _deserialize_layout(json.loads(obj_str))
 
 
-def deserialize_abstract_register(obj_str: str) -> BaseRegister:
+@overload
+def deserialize_abstract_register(
```

Comment:
All of these overload: pass confuse me.
Is that just the way to do it, or is there a reason for doing it if not required?

Reply:
They are only for `mypy`, they allow it to restrict the return type based on the type (or in this case the actual value) of the input. Without these, I'd have to `cast` the outputs to `Register` or `Register3D` on the `.from_abstract_repr()`

Reply:
Thanks!